@RequiresNonNull does not appear to be honored inside anonymous class

I would expect the following to check successfully:

MonotonicNonNullFieldTest.java

```
import org.checkerframework.checker.nullness.qual.*;

public class MonotonicNonNullFieldTest {
  class Data {
    @MonotonicNonNull Object field;
  }

  void method(Object object) {
  }

  @RequiresNonNull("#1.field")
  void test(final Data data) {
    method(data.field);  // checks OK

    Runnable callback = new Runnable() {
      public void run() {
        method(data.field);  // fails check
      }
    };
  }
}
```

However I get the following error:

```
$ java -jar checker.jar -processor org.checkerframework.checker.nullness.NullnessChecker -version MonotonicNonNullFieldTest.java
javac 1.8.0-jsr308-1.9.10
MonotonicNonNullFieldTest.java:17: error: [argument.type.incompatible] incompatible types in argument.
        method(data.field);  // fails check
                   ^
  found   : @Initialized @MonotonicNonNull Object
  required: @Initialized @NonNull Object
1 error
```

I saw https://github.com/typetools/checker-framework/issues/294 and https://github.com/typetools/checker-framework/issues/266, which appear to be describing a similar issue, but this issues seems different because it doesn't appear to depend on flow analysis.

--- a/checker/tests/nullness/MonotonicNonNullFieldTest.java
+++ b/checker/tests/nullness/MonotonicNonNullFieldTest.java
@@ -1,0 +1,23 @@
+// Testcase for Issue553
+// https://github.com/typetools/checker-framework/issues/553
+import org.checkerframework.checker.nullness.qual.*;
+
+public class MonotonicNonNullFieldTest {
+    class Data {
+        @MonotonicNonNull Object field;
+    }
+
+    void method(Object object) {}
+
+    @RequiresNonNull("#1.field")
+    void test(final Data data) {
+        method(data.field); // checks OK
+
+        Runnable callback =
+                new Runnable() {
+                    public void run() {
+                        method(data.field); // used to issue error
+                    }
+                };
+    }
+}

